### ISSUE TITLE
Problematic `seek(0)` in `simpleion.load_python`

### ISSUE THREAD

**Issue by melsabagh** [amazon-ion/ion-python#251]
The code below assumes the `fp` stream was at position `0` before reading the header:

https://github.com/amazon-ion/ion-python/blob/614a93fe4b15cf240356e9ff1b68ae49f3ffa37c/amazon/ion/simpleion.py#L361-L362

This behavior is quite unexpected and problematic as it forces the stream to rewind (may not be possible for network and custom streams) and also messes up any stream management logic performed by the caller. 

A quick fix would be:

```python
        pos = fp.tell()
        maybe_ivm = fp.read(4)
        fp.seek(pos)
```

**Comment by popematt** [comment]
Thanks for bring this to our attention. Are you interested in contributing a PR for this issue?

**Comment by melsabagh** [comment]
@popematt I opened a PR: https://github.com/amazon-ion/ion-python/pull/261

**Comment by jobarr-amzn** [comment]
Fixed by #261
